code/modules/wiremod/components/abstract/compare.dm

```diff
@@ -0,0 +1,69 @@
+/**
+ * # Compare Component
+ *
+ * Abstract component to build conditional components
+ */
+/obj/item/circuit_component/compare
+	display_name = "Compare"
+
+	/// The amount of input ports to have
+	var/input_port_amount = 4
+
+	/// The trigger for the true/false signals
+	var/datum/port/input/compare
+
+	/// Signals sent on compare
+	var/datum/port/output/true
+	var/datum/port/output/false
+
+	/// The result from the output
+	var/datum/port/output/result
+
+	var/result_type = PORT_TYPE_NUMBER
```

Comment:
Seems useless? Only used on line 35 but you could just put the definition itself on 35

Reply:
result_type was originally used for logic gates so that they could output any type, but I figured it had too many problems to work.